spec/models/watched_encode_spec.rb

```diff
@@ -45,12 +50,28 @@
       # Return a completed encode so the polling job doesn't run forever.
       allow(described_class.engine_adapter).to receive(:create).and_return(running_encode)
       allow(described_class.engine_adapter).to receive(:find).and_return(completed_encode)
+      # allow(master_file).to receive(:update_progress_on_success!)
     end
 
     it 'stores the encode id on the master file' do
-      allow(master_file).to receive(:update_progress_on_success!)
       encode.create!
       expect(master_file).to have_received(:update_progress_on_success!)
     end
+
+    context 'using Minio and ffmpeg adapter' do
+      before do
+        Settings.minio = double("minio", endpoint: "http://minio:9000", public_host: "http://domain:9000")
+        allow(Settings).to receive(:encoding).and_return(double(engine_adapter: "ffmpeg", derivative_bucket: "derivs"))
+        allow(File).to receive(:delete)
+      end
+
+      it 'uploads to Minio' do
+        encode.create!
```

Comment:
Are there any expectations for this test?  If it is just that it doesn't raise errors then I think it could be:
```
expect { encode.create! }.not_to raise_error
```